server/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -1793,7 +1793,9 @@ public Number value(XContentParser parser) throws IllegalArgumentException, IOEx
      */
     public void indexValue(DocumentParserContext context, Number numericValue) {
         if (dimension && numericValue != null) {
-            context.getDimensions().addLong(fieldType().name(), numericValue.longValue());
+            context.getDocumentFields().addLongDimension(fieldType().name(), numericValue.longValue());
+        } else if (fieldType().getMetricType() != null) {
```

Comment:
Right now histogram are not supported as metrics because we (still) don't know how aggregations need to work with them. What is `sum` or `avg` or any other aggregation going to do on a histogram?

Reply:
> we (still) don't know how aggregations need to work with them

I don't quite understand. What's the difference between doing any of these aggregations on a histogram field in a regular data stream vs a time series data stream?

Reply:
It has to do with the data that an histogram stores...are buckets cumulative or not? Are histograms Prometheus histograms or the result of using some library in integrations (tdigest or hdr)...without knowing the exact type of histograms those aggregations are likely to return garbage.

Reply:
I understand that your next question might be "but we already have them working on existing histogram fields...what's the difference?"

Reply:
The answer is that originally the histogram type was introduced for different use cases (don't know exactly all of them). Now we have new use cases due to integration of observability/metrics/... integrations which were not taken into account when the histogram field was introduced.